hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/balanceof/BalanceOfCall.java

```diff
@@ -67,36 +62,35 @@ public static boolean matches(@NonNull final byte[] selector) {
      * @return the constructed {@link BalanceOfCall}
      */
     public static BalanceOfCall from(@NonNull final HtsCallAttempt attempt) {
-        final var owner = fromHeadlongAddress(
-                BALANCE_OF.decodeCall(attempt.input().toArrayUnsafe()).get(0));
+        final Address owner =
+                BALANCE_OF.decodeCall(attempt.input().toArrayUnsafe()).get(0);
```

Comment:
So this `decodeCall` can throw if the `input` byte array is too short or too long (or otherwise doesn't decode).  But I traced the callers back up to `CustomMessageCallProcessor.start` and there are no exception handlers, though there is stuff dealing with manipulating and finishing frames and such.  So ... is there a handler somewhere that fixes things up?

Reply:
Fixed `HtsCallAttemptFactory` to catch exceptions and halt with `INVALID_OPERATION`, consuming all remaining gas.

Reply:
(Actually it is in `HtsSystemContract` now, TY.)